nameserver/server.go

```diff
@@ -194,17 +206,6 @@ func (s *DNSServer) Start() error {
 		Info.Printf("[dns] Cache: %d entries", s.cache.Capacity())
 	}
 
-	pc, err := net.ListenPacket("udp", s.ListenAddr)
-	if err != nil {
-		return err
-	}
-	s.pc = pc
-
-	_, port, err := net.SplitHostPort(pc.LocalAddr().String())
-	if err != nil {
-		return err
-	}
-	s.ListenAddr = fmt.Sprintf(":%s", port)
```

Comment:
Is this change really necessary? Do we need to move the `Listen()` to the `New()` method? I've never been fond of mixing strict initialization tasks (usually performed in `New()` methods, like memory allocations, checking parameters, etc...) with things that have side effects (an are usually delayed until we really `Start()`). I think we do not call `Start()` from multiple threads, do we?